src/main/java/net/minecraftforge/client/model/ForgeBlockStateV1.java

```diff
@@ -467,6 +467,58 @@ private static TRSRTransformation leftify(TRSRTransformation transform)
                 return TRSRTransformation.blockCenterToCorner(flipX.compose(TRSRTransformation.blockCornerToCenter(transform)).compose(flipX));
             }
 
+            // Note: these strings might change to a full-blown resource locations in the future, and move from here to some json somewhere
+            // TODO: vanilla now includes from parent, deprecate?
+            private static final ImmutableMap<String, IModelState> transforms;
+
+            static
```

Comment:
Is there a good place to call this so that it's not done in a `static` block?
It's not a huge problem but all the static initialization in Minecraft can make things a pain in unexpected ways sometimes and I'd like to avoid using the pattern more.

Reply:
I think it's hard to do differently while still keeping things static, final and immutable.
The transform stuff could be moved into a seperate class so it can be loaded seperately, perhaps?

Reply:
Hm right. In this case it's not likely to crash and doesn't load in lots of classes so I guess it's fine.

Reply:
"// TODO: vanilla now includes from parent, deprecate?" ?

Reply:
this code was just moved, the todo was added by @RainWarrior and I'm not sure what he meant:
https://github.com/MinecraftForge/MinecraftForge/commit/a799bbad2d2073400c5c9f327b4a07fe8dbe13da

Reply:
the meaning of that todo is:

in vanilla model jsons, the `display` block defining these transforms used to not be inherited from parents, meaning the default transforms had to be copy-pasted into every single item and model json. a hassle, so forge blockstate json lets you apply the default transforms to your model right in the blockstate json

sometime in 1.9, Mojang let models inherit `display` from their parents, meaning all blocks simply define `block/block` as their parent and they get all the transforms for free.

I imagine what fry meant was since it's no longer a hassle to get proper transforms in item and block models, to remove `forge:builtin-item` and their friends.

whether that's actually a good idea is questionable, since OBJ models and custom model loaders might not have the same support for inherited transforms